crates/matrix-sdk/src/event_cache/room/mod.rs

```diff
@@ -620,6 +620,79 @@ mod private {
             Ok(Self { room: room_id, store, events, waited_for_initial_prev_token: false })
         }
 
+        /// Load more events backwards if the last chunk is **not** a gap.
+        ///
+        /// Return `Ok(Some((events, reached_start), _))` if events have been
+        /// inserted, `Ok(None)` if a gap has been inserted or if the
+        /// store is disabled.
+        #[must_use = "Updates as `VectorDiff` must probably be propagated via `RoomEventCacheUpdate`"]
+        pub(in super::super) async fn load_more_events_backwards(
+            &mut self,
+        ) -> Result<
+            Option<((Vec<TimelineEvent>, bool), Vec<VectorDiff<TimelineEvent>>)>,
+            EventCacheError,
+        > {
+            // ⚠️ Propagate all pending updates before we do anything.
+            self.propagate_changes().await?;
```

Comment:
:warning: Why do we need to do this? It should be performed just after applying any changes to the in-memory representation of the linked chunk, and having to do it here would suggest there's a bug *elsewhere* (in the place that should've called `propagate_changes()` but didn't.

Strongly think we should not have to do this, and thus remove it.

Reply:
As discussed: I've not added this due to a bug, but by precaution. I will remove it and see what happens. It should not change anything, it was pure precaution.